src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/api/CardApiTest.java

```diff
@@ -157,4 +159,88 @@ void hide_correct_cards() {
         .jsonPath("_embedded.matches[*].rounds[0].cards[0].cardSuit")
         .isEqualTo(CLUB.name());
   }
+
+  @Test
+  void accept_card_playing() {
+
+    HttpHeaders responseHeaders =
+        webTestClient
+            .post()
+            .uri("/players")
+            .body(BodyInserters.fromValue(PLAYER_1))
+            .exchange()
+            .expectStatus()
+            .isCreated()
+            .expectBody()
+            .returnResult()
+            .getResponseHeaders();
+
+    String sessionIdPlayer1 = getSessionIdOf(responseHeaders);
+
+    responseHeaders =
+        webTestClient
+            .post()
+            .uri("/players")
+            .body(BodyInserters.fromValue(PLAYER_2))
+            .exchange()
+            .expectStatus()
+            .isCreated()
+            .expectBody()
+            .returnResult()
+            .getResponseHeaders();
+
+    String sessionIdPlayer2 = getSessionIdOf(responseHeaders);
+
+    Player player1 =
+        playerRepository.findAll().stream()
+            .filter(player -> player.getName().equals(PLAYER_NAME_1))
+            .findFirst()
+            .orElseThrow();
+
+    Player player2 =
+        playerRepository.findAll().stream()
+            .filter(player -> player.getName().equals(PLAYER_NAME_2))
+            .findFirst()
+            .orElseThrow();
+
+    Game game =
+        GameBuilder.builder("game1", gameRepository, participationRepository, playerRepository)
+            .withParticipationWith(player1)
+            .withParticipationWith(player2)
+            .withGameState(GameState.PLAYING)
+            .withMatch()
+            .withMatchState(MatchState.PLAYING)
+            .withHandForPlayer(PLAYER_NAME_1)
+            .withCards(ACE_OF_CLUBS, QUEEN_OF_CLUBS)
+            .finishHand()
+            .withRound()
+            .finishRound()
+            .finishMatch()
+            .build();
+
+    gameRepository.saveAll(List.of(game));
+    Card card =
+        cardRepository.findAll().stream()
+            .filter(c -> c.getRound() == null)
+            .filter(c -> c.getCardRank().equals(CardRank.ACE))
+            .findAny()
+            .orElseThrow();
+    Round round = game.getLastMatch().orElseThrow().getLastRound().orElseThrow();
+    card.setRound(round);
+
+    String uri = "cards/" + card.getId().toString() + "/round";
+
+    webTestClient
+        .patch()
+        .uri(uri)
+        .contentType(MediaType.APPLICATION_JSON)
+        .header(HttpHeaders.COOKIE, "JSESSIONID=%s".formatted(sessionIdPlayer1))
+        .body(BodyInserters.fromValue(round))
+        .exchange()
+        .expectStatus()
+        .is4xxClientError();
+    // .isOk();
+    // This test returns 405 METHOD_NOT_ALLOWED instead of Ok status and I have no idea why. Anyone
+    // any idea?
```

Comment:
```suggestion
    String uri = "cards/" + card.getId().toString();

    webTestClient
        .patch()
        .uri(uri)
        .contentType(MediaType.APPLICATION_JSON)
        .header(HttpHeaders.COOKIE, "JSESSIONID=%s".formatted(sessionIdPlayer1))
        .body(BodyInserters.fromValue(Map.of("round", "/rounds/%s".formatted(round.getId()))))
        .exchange()
        .expectStatus()
        .isOk();
```

If you fix all this, you will have a 422, you will find out why.

And after you fixed that, you will have a problem with serializing the card.

This piece of code will help:

```
        if (Card.class.isAssignableFrom(beanDesc.getBeanClass())) {
          if (serializer instanceof BeanSerializer) {
            return serializer;
          }
          return cardSerializer;
        }
```